### PR TITLE
[HTTP] Introduce common HTTP client config to remove repetition 

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/21049

## Approach
As mentioned in this https://github.com/ballerina-platform/ballerina-lang/issues/21049#issuecomment-590923455;

> List the copied fields in the referencing type and say "copied from T" with a link to T

## Remarks
- https://github.com/ballerina-platform/ballerina-spec/issues/423
- https://github.com/ballerina-platform/ballerina-lang/issues/21082

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [x] Module documentation in Module.md files
  - [x] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/ballerina/src/http/common_client_config.bal [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chamil321) [inline stdlib/http/src/main/ballerina/src/http/common_client_config.bal]: Shall we remove this content to http_commons.bal file as it contains all the commonly used types?

File: stdlib/http/src/main/ballerina/src/http/resiliency/failover_client_endpoint.bal [hunk 2]
```diff
@@ -434,37 +434,27 @@ function populateErrorsFromLastResponse (Response inResponse, ClientError?[] fai
...
 # + intervalInMillis - Failover delay interval in milliseconds
 public type FailoverClientConfiguration record {|
-    string httpVersion = HTTP_1_1;
-    ClientHttp1Settings http1Settings = {};
-    ClientHttp2Settings http2Settings = {};
-    int timeoutInMillis = 60000;
-    string forwarded = "disable";
-    FollowRedirects? followRedirects = ();
+    *CommonClientConfiguration;
     ProxyConfig? proxy = ();
```
Comment (chamil321) [inline stdlib/http/src/main/ballerina/src/http/resiliency/failover_client_endpoint.bal]: We can remove ProxyConfig as well as it is already grouped under ClientHttp1Settings type

File: stdlib/http/src/main/ballerina/src/http/resiliency/load_balance_client_endpoint.bal [hunk 3]
```diff
@@ -304,37 +304,27 @@ function populateGenericLoadBalanceActionError(LoadBalanceActionErrorData loadBa
...
 # + failover - Configuration for load balancer whether to fail over in case of a failure
 public type LoadBalanceClientConfiguration record {|
-    string httpVersion = HTTP_1_1;
-    ClientHttp1Settings http1Settings = {};
-    ClientHttp2Settings http2Settings = {};
-    int timeoutInMillis = 60000;
-    string forwarded = "disable";
-    FollowRedirects? followRedirects = ();
+    *CommonClientConfiguration;
     ProxyConfig? proxy = ();
```
Comment (chamil321) [inline stdlib/http/src/main/ballerina/src/http/resiliency/load_balance_client_endpoint.bal]: We can remove ProxyConfig as well as it is already grouped under ClientHttp1Settings type

File: language-server/modules/langserver-core/src/test/resources/completion/function/chainCompletion4.json [hunk 4]
```diff
@@ -4,358 +4,336 @@
...
-          "value": "**Package:** _ballerina/lang.array_  \n  \nReturns a stream from the given array.\n  \n"
-        }
-      },
-      "sortText": "120",
-      "insertText": "toStream()",
-      "insertTextFormat": "Snippet"
+    "sortText": "120",
+    "insertText": "shift()",
+    "insertTextFormat": "Snippet"
+  }, {
```
Comment (chamil321) [inline language-server/modules/langserver-core/src/test/resources/completion/function/chainCompletion4.json]: Shall we format the JSON using https://jsonformatter.curiousconcept.com/ and paste it here. Ideally, we need to add only the specific change. Most of these changes are just indentation.

Comment (BuddhiWathsala) [reply language-server/modules/langserver-core/src/test/resources/completion/function/chainCompletion4.json]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21275/commits/44296d135055b7826de9430384aaec4b1cbd9270

File: stdlib/http/src/main/ballerina/src/http/resiliency/load_balance_client_endpoint.bal (hunk 3 above)
Comment (BuddhiWathsala) [reply stdlib/http/src/main/ballerina/src/http/resiliency/load_balance_client_endpoint.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21275/commits/2bbf173f2e61c966f125812b80ddcdaf81b4ccff

File: stdlib/http/src/main/ballerina/src/http/resiliency/failover_client_endpoint.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply stdlib/http/src/main/ballerina/src/http/resiliency/failover_client_endpoint.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21275/commits/2bbf173f2e61c966f125812b80ddcdaf81b4ccff

File: stdlib/http/src/main/ballerina/src/http/common_client_config.bal (hunk 1 above)
Comment (BuddhiWathsala) [reply stdlib/http/src/main/ballerina/src/http/common_client_config.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21275/commits/2bbf173f2e61c966f125812b80ddcdaf81b4ccff

**Pull request merged**